### ISSUE TITLE
[$8000] Desktop - A red dot in the LHN appears next to a chat

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14679]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Log out of New.Expensify
2. Log back in to New.Expensify
3. Send a message to someone
4. See a red dot next to the chat

_Note: we are currently unable to provide a reliable set of reproduction steps, as this can occur even without logging in or out. All we've established is that the dot seems to appear next to a person you've had a conversation history with._

## Expected Result:
This red dot shouldn't currently exist for any contact in the LHN

## Actual Result:
A red dot appears in the LHN

## Workaround:
Unknown, as there doesn't seem to be any consistent behaviour that generates this red dot.

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:** Yes - v1.2.66. However, needs consistent reproduction
**Reproducible in production?:** Yes - However, needs consistent reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
<img width="1440" alt="image (2)" src="https://user-images.githubusercontent.com/43996225/215636157-0b9d8190-6b73-43dc-bb43-5402c5e91985.png">

![image](https://user-images.githubusercontent.com/30815269/215674556-390a4911-f87b-4ca9-a484-f54e976935af.png)

![image](https://user-images.githubusercontent.com/30815269/215674580-45429f10-5f7f-401b-b54e-51aa3479384d.png)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1674781889249199

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01aeb9ca8d306762f0</li>
        <li>Upwork Job ID: 1620299417834770432</li>
        <li>Last Price Increase: 2023-02-21</li>
    </ul>
</details>

**Comment by jliexpensify** [comment]
Did a nunch of repro-ing and have a theory, posted here: https://expensify.slack.com/archives/C049HHMV9SM/p1675131225201399?thread_ts=1674781889.249199&cid=C049HHMV9SM

**Comment by jliexpensify** [comment]
Here's another [issue](https://github.com/Expensify/App/issues/14570) with the same red dot, but it appears when a bill is split.

**Comment by Elkaramany** [comment]
**Proposal**

Remove the Expensicons.DotIndicator Icon from the OptionRowLHN component. I think this may work by removing the following piece of code: 

` {optionItem.brickRoadIndicator === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR && (
                                    <View style={[styles.alignItemsCenter, styles.justifyContentCenter]}>
                                        <Icon
                                            src={Expensicons.DotIndicator}
                                            fill={colors.red}
                                            height={variables.iconSizeSmall}
                                            width={variables.iconSizeSmall}
                                        />
                                    </View>
                                )}`

**Comment by Elkaramany** [comment]
https://github.com/Expensify/App/blob/f624d828408543e64422c18a5156ec3e493c0253/src/components/LHNOptionsList/OptionRowLHN.js#L202

**Comment by Elkaramany** [comment]
https://github.dev/Expensify/App/blob/main/src/components/LHNOptionsList/OptionRowLHN.js#L202-L211

**Comment by youssef-lr** [comment]
I've just encountered this error in dev but when trying to settling up an IOU. I can see in the logs what the error is and for me it was a missing `clientID`. As far as I know, with the deprecation of sequence numbers we don't need those anymore and we shouldn't be checking for them. I will confirm this in Slack and report back here. I'm still testing various scenarios.

**Comment by sobitneupane** [comment]
Thanks for your [proposal](https://github.com/Expensify/App/issues/14679#issuecomment-1410215346) @Elkaramany.

But we don't want to completely get rid of DotIndicators. DotIndicators are required in case of an error.

**Comment by Elkaramany** [comment]
@sobitneupane No problem, can you please elaborate on the cases required to achieve this: This red dot shouldn't currently exist for any contact in the LHN. Thanks.

**Comment by jliexpensify** [comment]
@youssef-lr - just wondering: would you want to take this/should it be an Internal issue?

**Comment by youssef-lr** [comment]
I'm not sure yet if this could be internal only...I'm not able to reproduce it anymore. Is it still occurring on your end?

**Comment by jliexpensify** [comment]
@youssef-lr yep, still appearing on v1.2.63-0

Specifically with Kadie - she seems to be the only person I see the dot with:

![image](https://user-images.githubusercontent.com/30815269/216176996-f9a4b895-8060-4e73-bd2e-d97ff2ae3851.png)

**Comment by youssef-lr** [comment]
Hmm so there is an IOU involved there. Have you had this error happen in a chat with no IOUs?

**Comment by jliexpensify** [comment]
My original theory was it had something to do with the Wallet, but I asked someone in the US to test, and another person who hasn't requested money from me. [I added thoughts in this thread](https://expensify.slack.com/archives/C049HHMV9SM/p1674781889249199) so I don't think it's necessarily to do with IOU's.

**Comment by jliexpensify** [comment]
@youssef-lr so I was chatting to @bondydaa and noticed that the red dot suddenly appeared again (I've never requested money from Bondy):

![image](https://user-images.githubusercontent.com/30815269/217107072-c28a1255-29a1-4b27-a0ca-946641191cdb.png)

On Staging v1.2.66-0

**Comment by jliexpensify** [comment]
Hi @youssef-lr - any luck reproducing on your side?

**Comment by youssef-lr** [comment]
@jliexpensify I'm sadly unable to reproduce this.

**Comment by jliexpensify** [comment]
@youssef-lr - I spent some time testing this morning, and noticed that the "decline money" button wasn't working with Kadie. After a few attempts, I was able to finally decline and noticed this error message surfaced:

![image](https://user-images.githubusercontent.com/30815269/218345202-c24b44ec-6a78-457f-8a6f-069d58bdb8b4.png)

I was then actually able to get rid of the red dot next to Kadie's name. Steps taken:

1. Scrolled up to the error message in my chat with Kadie
2. On the right-hand side, there was a "X" next to the message, so I clicked it
3. Confirmed the error message disappeared
4. However, the dot didn't disappear from Kadie's name, so I logged out and logged in
5. Logged in and the dot now does not show next to Kadie's name

I didn't take a screenshot, but the "X" was here:

![image](https://user-images.githubusercontent.com/30815269/218345616-238e6acc-bd87-4fc7-a3e2-64ba946ac434.png)

Does any of that help @youssef-lr @sobitneupane ? This is a weird error - my understanding is that the red dot is surfaced when there's an error with the chat (in my case, the inability to decline money) but I've noticed it on other chats where I don't think there have been any errors/no money has been sent.

**Comment by MelvinBot** [comment]
@youssef-lr @sobitneupane @jliexpensify this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by jliexpensify** [comment]
See notes above Melvin! cc @youssef-lr

**Comment by sobitneupane** [comment]
Tried to reproduce the issue. But no luck.

**Comment by jliexpensify** [comment]
Yeah, I don't know what to do about this: my theory is that the dot appears next to a chat that has an issue (in my case with Kadie, a failed payment cancellation and failed sending messages), but it's also started showing in chats where everything seems fine. So I also can't reliably reproduce, and the above troubleshooting is the closest I've come to eliminating the dot. I've also chatted to quite a few colleagues and can't pin down the exact behaviours that cause the dot to appear/disappear.

@youssef-lr what do you think we should do? I'm happy to take it to the channel but I don't know if we'll get anywhere as no-one seems to be able to find the cause/reliably reproduce it.

**Comment by sobitneupane** [comment]
I was able to [reproduce](https://github.com/Expensify/App/pull/14736#issuecomment-1415384081) similar issue few weeks back while reviewing [this PR](https://github.com/Expensify/App/pull/14736). But I can no longer reproduce it.

@aldo-expensify had [mentioned](https://github.com/Expensify/App/pull/14736#issuecomment-1416233880) it was being handled by another PR for the backend.

https://user-images.githubusercontent.com/25876548/219269016-c190e9db-ac9b-4d81-95cd-eca001650c3b.mov

**Comment by jliexpensify** [comment]
I signed out and back in and checked a few chats that previously had red dots, and they're all gone! That matches with the deployment of that PR (I was OOO yesterday) - so this might be fixed? 

I'm also checking with a few colleagues to see if they're seeing red dots.

**Comment by jliexpensify** [comment]
Ok yeah - a colleague who was seeing a red dot next to my name has reported it's now gone! Going to assume that Aldo's PR fixed this issue - would you say that's correct @sobitneupane ?

**Comment by sobitneupane** [comment]
I believe so. I would love to hear from @aldo-expensify. He might have more context about the PR.

**Comment by aldo-expensify** [comment]
This is the PR in the backend fixing it: https://github.com/Expensify/Web-Expensify/pull/36352 . It was because we were still using the `clientID` instead of the `reportActionID` in some error handling code.

This bug looks very very similar. Maybe there is some place left to update.

There is this PR https://github.com/Expensify/Web-Expensify/pull/36504 from @tgolen that is removing the `clientID` in several more places. Maybe this bug will get fixed once this PR is merged.

**Comment by jliexpensify** [comment]
Happy to keep this open until the 2nd PR gets merged, but I got some responses from colleagues overnight who also experienced the red dot issue, and a log out/log in seems to have resolved it.

**Comment by jliexpensify** [comment]
p.s. Thanks Aldo - this was a super frustrating issue to repro!

**Comment by aldo-expensify** [comment]
I think if we take a look at the Onyx DB on an account that is experiencing the problem we may find a clue on where the error is being set and we may see a clientID involved

**Comment by aldo-expensify** [comment]
so.. if you see the error again, don't log out/log in and lets try to inspect Onyx

**Comment by MelvinBot** [comment]
@youssef-lr, @sobitneupane, @jliexpensify Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by jliexpensify** [comment]
Not overdue, but given the red dot isn't showing anymore, how do you feel about closing this @youssef-lr @aldo-expensify ?

**Comment by aldo-expensify** [comment]
I agree with closing it and reopen if it happens again.

Remember that if it happens again, don't fix it and lets try to check the data in Onyx.

**Comment by aldo-expensify** [comment]
Left it open for @youssef-lr to make the final decision since he is assigned

**Comment by MelvinBot** [comment]
@youssef-lr @sobitneupane @jliexpensify this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MelvinBot** [comment]
@youssef-lr, @sobitneupane, @jliexpensify Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by jliexpensify** [comment]
Bump @youssef-lr - are we good to close this issue, based off Aldo's recommendation?

>I agree with closing it and reopen if it happens again.

**Comment by youssef-lr** [comment]
Sounds good to me, I had a feeling this was about `clientIDs` [here](https://github.com/Expensify/App/issues/14679#issuecomment-1410759151) as well so I think that PR is going to fix this. Closing for now!
